targets/TARGET_STM/sleep.c

```diff
@@ -161,6 +161,7 @@ void hal_deepsleep(void)
     // Disable IRQs
     core_util_critical_section_enter();
 
+    // Save the timer counter value in order to reprogram it after deepsleep
```

Comment:
I would prefer a new function in hal_tick_common.c which saves TIM_MST needed registers before going into deepsleep:
- CNT
- CCR1
- DIER

Reply:
done in commit https://github.com/ARMmbed/mbed-os/pull/7352/commits/a838cb21d12473cf5cd29c708bbd8f381f8d37f7